### ISSUE TITLE
`Lint/Void` - undefined method `name` on `node.loc.name`

### ISSUE THREAD

**Issue by snutij** [rubocop/rubocop#12002]
Hello 👋  Tell me if more information is needed 🙏 

Given a file `foo.rb`: 
```ruby
__ENCODING__
__FILE__


```

Which is a valid syntax: 

```
ruby -c foo.rb
Syntax OK
```

--------

## Expected behavior

Should not trigger an error, and to what I understand no offense should be displayed, but I can be wrong on that.

## Actual behavior

`bin/rubocop -d --only Lint/Void foo.rb`
 
```
An error occurred while Lint/Void cop was inspecting foo.rb:1:0.
undefined method `name' for #<Parser::Source::Map:0x00000001052d22f0 @expression=#<Parser::Source::Range foo.rb 0...12>, @node=s(:const,
  s(:const, nil, :Encoding), :UTF_8)>
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:117:in `check_var'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:96:in `check_expression'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:90:in `block in check_begin'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:90:in `each'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:90:in `check_begin'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:81:in `on_begin'
/Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/commissioner.rb:107:in `public_send'
```

## Steps to reproduce the problem

In `RuboCop::Cop::Lint::Void` the method `check_var` is trigger, 

Which receive a node: 

```ruby
(rdbg) node
s(:const,
  s(:const, nil, :Encoding), :UTF_8)

(ruby) node.source
"__ENCODING__"
```

There is no early return because: 

```
(ruby) node.const_type?
true
```

Then in `add_offense` we search for `node.location.name` that trigger the error:

```ruby
(ruby) node.loc
#<Parser::Source::Map:0x000000010963d9e0
 @expression=#<Parser::Source::Range foo.rb 0...12>,
 @node=s(:const,
  s(:const, nil, :Encoding), :UTF_8)>

(ruby) node.loc.name
eval error: undefined method `name' for #<Parser::Source::Map:0x000000010963d9e0 @expression=#<Parser::Source::Range foo.rb 0...12>, @node=s(:const,
  s(:const, nil, :Encoding), :UTF_8)>
  (rdbg)//Users/snutij/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.53.0/lib/rubocop/cop/lint/void.rb:1:in `check_var'
nil
```

https://github.com/rubocop/rubocop/blob/4f83077d045135b411dca3f59cfd5c8290ab5322/lib/rubocop/cop/lint/void.rb#L113-L120

## RuboCop version

```
bundle exec rubocop -V
1.53.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
  - rubocop-minitest 0.31.0
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.0
```
